builder/amazon/common/step_get_password.go

```diff
@@ -119,11 +120,25 @@ func (s *StepGetPassword) waitForPassword(state multistep.StateBag, cancel <-cha
 		case <-time.After(5 * time.Second):
 		}
 
-		resp, err := ec2conn.GetPasswordData(&ec2.GetPasswordDataInput{
-			InstanceId: instance.InstanceId,
+		// Wrap in a retry so that we don't fail on rate-limiting.
+		log.Printf("Retrieving auto-generated instance password...")
+		var resp *ec2.GetPasswordDataOutput
+		err := retry.Config{
+			Tries:      11,
+			RetryDelay: (&retry.Backoff{InitialBackoff: 200 * time.Millisecond, MaxBackoff: 30 * time.Second, Multiplier: 2}).Linear,
+		}.Run(ctx, func(ctx context.Context) error {
+			var err error
+			resp, err = ec2conn.GetPasswordData(&ec2.GetPasswordDataInput{
```

Comment:
here, if we use `GetPasswordDataRequest` and them run `SetContext` on the request; it will be cancelled by the context.

```suggestion
			resp = ec2conn.GetPasswordDataRequest(&ec2.GetPasswordDataInput{
```